src/plugins/data_views/server/register_index_pattern_usage_collection.ts

```diff
@@ -80,60 +85,74 @@ export async function getIndexPatternTelemetry(indexPatterns: DataViewsContract)
+      const fields = JSON.parse(obj.attributes.fields) || [];
+      const runtimeFieldsMap = obj.attributes.runtimeFieldMap
+        ? JSON.parse(obj.attributes.runtimeFieldMap) || {}
+        : {};
+      const scriptedFields: FieldSpec[] = fields.filter((fld: FieldSpec) => !!fld.script);
+      // to array
+      const runtimeFields: RuntimeField[] = Object.values(runtimeFieldsMap);
 
       // calc LoC
-      const runtimeFieldScripts = runtimeFields.map(
-        (fld) => fld.runtimeField?.script?.source || ''
-      );
-      results.perIndexPattern.runtimeFieldLineCount = minMaxAvgLoC(runtimeFieldScripts);
-
-      // calc field counts
-      results.perIndexPattern.runtimeFieldCount.min = updateMin(
-        results.perIndexPattern.runtimeFieldCount.min,
-        runtimeFields.length
-      );
-      results.perIndexPattern.runtimeFieldCount.max = updateMax(
-        results.perIndexPattern.runtimeFieldCount.max,
-        runtimeFields.length
-      );
-      results.perIndexPattern.runtimeFieldCount.avg =
-        results.runtimeFieldCount / results.indexPatternsWithRuntimeFieldCount;
-    }
-  }, Promise.resolve());
+      if (scriptedFields.length > 0) {
+        // increment counts
+        results.indexPatternsWithScriptedFieldCount++;
+        results.scriptedFieldCount += scriptedFields.length;
+
+        // calc LoC
+        results.perIndexPattern.scriptedFieldLineCount = minMaxAvgLoC(
+          scriptedFields.map((fld) => fld.script || '')
+        );
+
+        // calc field counts
+        results.perIndexPattern.scriptedFieldCount.min = updateMin(
+          results.perIndexPattern.scriptedFieldCount.min,
+          scriptedFields.length
+        );
+        results.perIndexPattern.scriptedFieldCount.max = updateMax(
+          results.perIndexPattern.scriptedFieldCount.max,
+          scriptedFields.length
+        );
+        results.perIndexPattern.scriptedFieldCount.avg =
+          results.scriptedFieldCount / results.indexPatternsWithScriptedFieldCount;
+      }
+
+      if (runtimeFields.length > 0) {
+        // increment counts
+        results.indexPatternsWithRuntimeFieldCount++;
+        results.runtimeFieldCount += runtimeFields.length;
+
+        // calc LoC
+        const runtimeFieldScripts = runtimeFields.map((fld) => fld.script?.source || '');
+        results.perIndexPattern.runtimeFieldLineCount = minMaxAvgLoC(runtimeFieldScripts);
+
+        // calc field counts
+        results.perIndexPattern.runtimeFieldCount.min = updateMin(
+          results.perIndexPattern.runtimeFieldCount.min,
+          runtimeFields.length
+        );
+        results.perIndexPattern.runtimeFieldCount.max = updateMax(
+          results.perIndexPattern.runtimeFieldCount.max,
+          runtimeFields.length
+        );
+        results.perIndexPattern.runtimeFieldCount.avg =
+          results.runtimeFieldCount / results.indexPatternsWithRuntimeFieldCount;
+      }
+    });
+  }
 
   return results;
```

Comment:
console.log here to see results. results are encrypted in the http response. You may need to restart kibana to run this multiple times.